layer1/chains/ethereum/contracts.go

```diff
@@ -48,7 +48,7 @@ type Contracts struct {
 // Set the contractFactoryAddress and looks up for all the contracts that we
 // need that were deployed via the factory. It's only executed once. Other call
 // to this functions are no-op.
-func NewContracts(eth *Client, contractFactoryAddress common.Address) *Contracts {
+func NewContracts(eth *evm.Client, contractFactoryAddress common.Address) *Contracts {
```

Comment:
There is a lot in this function not covered by tests.

Reply:
Since I moved these files around and coverage looks at the diffs it spotted all the parts without tests. I think adding the tests to those files and raising coverage should be tackled in a different ticket. I didn't add any new code, just moving and renaming. This Pr is just so the next PR adding the polygon doesn't get 100+ files changing.